README.md
![alt text][logo]

[logo]: https://github.com/MagnumOpuses/project-meta/blob/master/img/jobtechdev_black.png "JobTech dev logo"

[A JobTech Project](https://www.jobtechdev.se)

# Jobtechdev.se (documentation website for developers)

This project's purpose is to serve all necessary details for developers to integrate with Jobtech Dev's open infrastructure. This is the central hub to find information about our infrastructure projects, API specifications, integration guidelines, licenses, contact details and other important notices.

The website is built using the static website generation framework [Hugo](https://gohugo.io/) (see [Hugo documentation](https://gohugo.io/documentation/)).

## Getting started

First and foremost, if you wish to develop for and contribute to this open-source project. Please read the guidelines for [Contributing](CONTRIBUTING.md) and the [Hugo Getting Started Guide](https://gohugo.io/getting-started/). This will provide you all required knowledge to develop for this project.

Next up, install all [Prerequisites](#prerequisites) and follow the [Installation](#installation) steps.

Local compilation and deployment is a breeze with the provided [NPM](https://www.npmjs.com) run scripts and Dockerfile build instructions (please see [Deployment](#deployment) section).

### Prerequisites

 * [Git](https://git-scm.com/)
 * [Hugo](https://gohugo.io/)
 * [Node.js](https://nodejs.org/en/)
 * [Docker](https://www.docker.com)

### Installation

 1. Clone this project, see guide: [Cloning a repository](https://help.github.com/en/articles/cloning-a-repository)
 1. Install all project dependencies with command:
    ```bash
    npm install
    ```

## Test

:warning: *No tests available*

## Deployment

During development on this project it's usually preferable to deploy directly to the local Hugo development server to fully benefit from the automatic [watch and live-reload](https://gohugo.io/getting-started/usage/#livereload) features.

### Deploy to local Hugo development server
```bash
npm run start
```

Visit the locally deployed website here: http://localhost:3000

### Deploy to local Docker environment

#### Stage docker image

```bash
docker build --build-arg ARG_BUILDNAME=stage -t <image name> -f Dockerfile .
docker run -it  -p 8080:8080 -p 80:8080 <image name>
```

Visit the locally deployed website here: http://localhost:8080
 * Username: `default`
 * Password: `default`

#### Production docker image

```bash
docker build -t <image name> -f Dockerfile .
docker run -it  -p 8080:8080 -p 80:8080 <image name>
```

Visit the locally deployed website here: http://localhost:8080

## Built with

* Any IDE or text editor is fine.
* Compiled with [Hugo](https://gohugo.io/).

## Contributing

We would love if you'd like to help us build and improve this product for the benefit of everyone. This project is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere to the [Contributor Covenant](http://contributor-covenant.org/) code of conduct.

Any contributions, feedback and suggestions are more than welcome.

Please read our guidelines for contribution [here](CONTRIBUTING_TEMPLATE.md).

## License

[Apache License 2.0](LICENSE.md)

## Acknowledgments

N/A


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]
None


## [1.1.3] - 2019

### Fixed
- nginx points 404 to /404.html
- Swedish cookie popup
- Swedish API status
- Minor cleaning of content

## [1.1.2] - 2019-07-15 

### Fixed
- Seperated Analytics and Tag Mangager accounts.
- Fixed Content Security spec to support font, analytics and tag manager.  

## [1.1.1] - 2019-07-15 

### Fixed
- Added Google Tag Manager script and tag

## [1.1.0] - 2019-07-08

### Fixed
- Mydata github links pointed to old repo fixed (https://github.com/MagnumOpuses/jobtechdev.se/pull/59)
- Broken link to AF Jobs Swagger UI (https://github.com/MagnumOpuses/jobtechdev.se/pull/56)
- Weights on most pages to avoid page order ambiguity in left-hand menu (https://github.com/MagnumOpuses/jobtechdev.se/pull/61)

### Removed
- Unnecessary/Duplicated Swagger files (https://github.com/MagnumOpuses/jobtechdev.se/pull/56)

# Changelog

## [1.0.1] - 2019-07-02

### Removed
- JobScanner preview links due to service being temporarily unavailable (https://github.com/MagnumOpuses/jobtechdev.se/pull/53)

## [1.0.0] - 2019-07-01

### Added
- Cookie policy and cookie banner on front page and documentation pages (https://github.com/MagnumOpuses/jobtechdev.se/pull/39, https://github.com/MagnumOpuses/jobtechdev.se/pull/49)
- Readme instructions on how to build/run site in 'stage'-mode (https://github.com/MagnumOpuses/jobtechdev.se/pull/23)
- New event 'Tech for lunch!' (https://github.com/MagnumOpuses/jobtechdev.se/pull/25)
- Information about closure of old YrkesInfo & RAF API (https://github.com/MagnumOpuses/jobtechdev.se/pull/24)
- 'Theos'-styled pull-request template (https://github.com/MagnumOpuses/jobtechdev.se/pull/22, https://github.com/MagnumOpuses/jobtechdev.se/pull/33)
- Extra information about JobScanner showcase (https://github.com/MagnumOpuses/jobtechdev.se/pull/50)
- Added new shortcodes to allow tabbed content in markup (https://github.com/MagnumOpuses/jobtechdev.se/pull/49)

### Fixed
- Revapi not defined error (https://github.com/MagnumOpuses/jobtechdev.se/pull/38)
- Menu sliding bug in IE (https://github.com/MagnumOpuses/jobtechdev.se/pull/35)
- Featherlight popup to be dark themed (https://github.com/MagnumOpuses/jobtechdev.se/pull/44)
- Jobtech Dev logo in repository README.md (https://github.com/MagnumOpuses/jobtechdev.se/pull/48)

### Changed
- Upgraded Hugo to v0.55 (https://github.com/MagnumOpuses/jobtechdev.se/pull/34)
- Upgraded Hugo-theme-learn to v2.3.0 (https://github.com/MagnumOpuses/jobtechdev.se/pull/34)
- Fixed typo in swedish navbar (https://github.com/MagnumOpuses/jobtechdev.se/pull/31)
- Darkened the bold text in notice boxes (https://github.com/MagnumOpuses/jobtechdev.se/pull/37)
- Swagger UI theme to better fit Jobtechdev's dark themed website (https://github.com/MagnumOpuses/jobtechdev.se/pull/18)
- Upgraded PDF.js to v2.1.266 (https://github.com/MagnumOpuses/jobtechdev.se/pull/42)

### Removed
- Removed 'Vacancies' showcase (https://github.com/MagnumOpuses/jobtechdev.se/pull/43)
- Disabled language switching button in Docs pages (https://github.com/MagnumOpuses/jobtechdev.se/pull/49)

### Security
- CSP rule to allow image sources from 'online.swagger.io' (https://github.com/MagnumOpuses/jobtechdev.se/pull/32)
- CSP rule to allow blob image sources from whitelisted domains (https://github.com/MagnumOpuses/jobtechdev.se/pull/42)


thebeginning/js/rev.slider.js
'use strict';

var dzrevapi;
var dzQuery =jQuery;
function dz_rev_slider_4(){
	'use strict';
	if(dzQuery("#dz_rev_slider_3").revolution == undefined){
		revslider_showDoubleJqueryError("#dz_rev_slider_4");
	}else{
		dzrevapi = dzQuery("#dz_rev_slider_4").show().revolution({
			sliderType:"standard",
			jsFileLocation:"//server.local/revslider/wp-content/plugins/revslider/public/assets/js/",
			sliderLayout:"fullscreen",
			dottedOverlay:"none",
			delay:9000,
			navigation: {
					keyboardNavigation: "off",
					keyboard_direction: "vertical",
					mouseScrollNavigation: "off",
					onHoverStop: "off",
					touch: {
						touchenabled: "off",
						swipe_threshold: 75,
						swipe_min_touches: 1,
						swipe_direction: "vertical",
						drag_block_vertical: false
					},
					arrows: {
						style: "erinyen",
						enable: true,
						hide_onmobile: false,
						hide_onleave: false,
						tmp: '<div class="tp-title-wrap">  	<div class="tp-arr-imgholder"></div>    <div class="tp-arr-img-over"></div>	<span class="tp-arr-titleholder">{{title}}</span> </div>',
						left: {
							h_align: "left",
							v_align: "center",
							h_offset: 10,
							v_offset: 0
						},
						right: {
							h_align: "right",
							v_align: "center",
							h_offset: 10,
							v_offset: 0
						}
					},
					
				},
			
			responsiveLevels:[1240,1024,778,480],
			visibilityLevels:[1240,1024,778,480],
			gridwidth:[1240,1024,778,480],
			gridheight:[700,700,700	,400],
			lazyType:"none",
			parallax: {
				type:"scroll",
				origo:"enterpoint",
				speed:400,
				levels:[5,10,15,20,25,30,35,40,45,50,46,47,48,49,50,55],

			},
			shadow:0,
			spinner:"off",
			stopLoop:"off",
			stopAfterLoops:-1,
			stopAtSlide:-1,
			shuffle:"off",
			autoHeight:"off",
			hideThumbsOnMobile:"off",
			hideSliderAtLimit:0,
			fullScreenOffset: "60px",
			hideCaptionAtLimit:0,
			hideAllCaptionAtLilmit:0,
			debugMode:false,
			fallbacks: {
				simplifyAll:"off",
				nextSlideOnWindowFocus:"off",
				disableFocusListener:false,
			}
		});
	}
}

function dz_rev_slider_1(){
	'use strict';
	if(dzQuery("#rev_slider_1175_1").revolution == undefined){
		revslider_showDoubleJqueryError("#rev_slider_1175_1");
	}else{
		dzrevapi = dzQuery("#rev_slider_1175_1").show().revolution({
			sliderType:"standard",
			jsFileLocation:"//server.local/revslider/wp-content/plugins/revslider/public/assets/js/",
			sliderLayout:"fullscreen",
			dottedOverlay:"none",
			delay:9000,
			navigation: {
				keyboardNavigation:"off",
				keyboard_direction: "horizontal",
				mouseScrollNavigation:"off",
				mouseScrollReverse:"default",
				onHoverStop:"off",
				touch:{
					touchenabled:"off",
					swipe_threshold: 75,
					swipe_min_touches: 1,
					swipe_direction: "horizontal",
					drag_block_vertical: false
				}
				,
				arrows: {
					style:"uranus",
					enable:true,
					hide_onmobile:true,
					hide_under:768,
					hide_onleave:false,
					tmp:'',
					left: {
						h_align:"left",
						v_align:"center",
						h_offset:20,
						v_offset:0
					},
					right: {
						h_align:"right",
						v_align:"center",
						h_offset:20,
						v_offset:0
					}
				}
			},
			responsiveLevels:[1240,1024,778,480],
			visibilityLevels:[1240,1024,778,480],
			gridwidth:[1240,1024,778,480],
			gridheight:[868,768,960,720],
			lazyType:"smart",
			parallax: {
				type:"scroll",
				origo:"slidercenter",
				speed:400,
				levels:[5,10,15,20,25,30,35,40,45,46,47,48,49,50,51,55],
				type:"scroll",
			},
			shadow:0,
			spinner:"off",
			stopLoop:"on",
			stopAfterLoops:0,
			stopAtSlide:1,
			shuffle:"off",
			autoHeight:"off",
			fullScreenAutoWidth:"off",
			fullScreenAlignForce:"off",
			fullScreenOffsetContainer: "",
			fullScreenOffset: "70",
			disableProgressBar:"on",
			hideThumbsOnMobile:"off",
			hideSliderAtLimit:0,
			hideCaptionAtLimit:0,
			hideAllCaptionAtLilmit:0,
			debugMode:false,
			fallbacks: {
				simplifyAll:"off",
				nextSlideOnWindowFocus:"off",
				disableFocusListener:false,
			}
		});
	}
}

function dz_rev_slider_2(){
	'use strict';
	if(dzQuery("#rev_slider_989_1").revolution == undefined){
		revslider_showDoubleJqueryError("#rev_slider_989_1");
	}else{
		dzrevapi = dzQuery("#rev_slider_989_1").show().revolution({
			sliderType:"hero",
			jsFileLocation:"//server.local/revslider/wp-content/plugins/revslider/public/assets/js/",
			sliderLayout:"fullscreen",
			dottedOverlay:"none",
			delay:9000,
			navigation: {
			},
			responsiveLevels:[1240,1024,778,480],
			visibilityLevels:[1240,1024,778,480],
			gridwidth:[1240,1024,778,480],
			gridheight:[868,768,500,720],
			lazyType:"none",
			parallax: {
				type:"scroll",
				origo:"slidercenter",
				speed:400,
				levels:[5,10,15,20,25,30,35,40,45,46,47,48,49,50,51,55],
				type:"scroll",
			},
			shadow:0,
			spinner:"off",
			autoHeight:"off",
			fullScreenAutoWidth:"off",
			fullScreenAlignForce:"off",
			fullScreenOffsetContainer: "",
			fullScreenOffset: "0px",
			disableProgressBar:"on",
			hideThumbsOnMobile:"off",
			hideSliderAtLimit:0,
			hideCaptionAtLimit:0,
			hideAllCaptionAtLilmit:0,
			debugMode:false,
			fallbacks: {
				simplifyAll:"off",
				disableFocusListener:false,
			}
		});
	}
}
function dz_rev_slider_3(){
	'use strict';
	if(dzQuery("#rev_slider_1077_1").revolution == undefined){
		revslider_showDoubleJqueryError("#rev_slider_1077_1");
	}else{
		dzrevapi = dzQuery("#rev_slider_1077_1").show().revolution({
			sliderType:"standard",
			jsFileLocation:"//server.local/revslider/wp-content/plugins/revslider/public/assets/js/",
			sliderLayout:"fullscreen",
			dottedOverlay:"none",
			delay:9000,
			navigation: {
				keyboardNavigation:"off",
				keyboard_direction: "horizontal",
				mouseScrollNavigation:"off",
				mouseScrollReverse:"default",
				onHoverStop:"off",
				touch:{
					touchenabled:"on",
					swipe_threshold: 75,
					swipe_min_touches: 1,
					swipe_direction: "horizontal",
					drag_block_vertical: false
				}
				,
				bullets: {
					enable:true,
					hide_onmobile:true,
					hide_under:960,
					style:"zeus",
					hide_onleave:false,
					direction:"horizontal",
					h_align:"right",
					v_align:"bottom",
					h_offset:80,
					v_offset:50,
					space:5,
					tmp:'<span class="tp-bullet-image"></span><span class="tp-bullet-imageoverlay"></span><span class="tp-bullet-title">{{title}}</span>'
				}
			},
			responsiveLevels:[1240,1024,778,480],
			visibilityLevels:[1240,1024,778,480],
			gridwidth:[1240,1024,778,480],
			gridheight:[868,768,960,720],
			lazyType:"none",
			parallax: {
				type:"mouse",
				origo:"slidercenter",
				speed:2000,
				levels:[2,3,4,5,6,7,12,16,10,50,46,47,48,49,50,55],
				type:"mouse",
				disable_onmobile:"on"
			},
			shadow:0,
			spinner:"off",
			stopLoop:"on",
			stopAfterLoops:0,
			stopAtSlide:1,
			shuffle:"off",
			autoHeight:"off",
			fullScreenAutoWidth:"off",
			fullScreenAlignForce:"off",
			fullScreenOffsetContainer: "",
			fullScreenOffset: "0px",
			disableProgressBar:"on",
			hideThumbsOnMobile:"off",
			hideSliderAtLimit:0,
			hideCaptionAtLimit:0,
			hideAllCaptionAtLilmit:0,
			debugMode:false,
			fallbacks: {
				simplifyAll:"off",
				nextSlideOnWindowFocus:"off",
				disableFocusListener:false,
			}
		});
		var newCall = new Object(),
		cslide;

		newCall.callback = function() { 
		var proc = revapi1077.revgetparallaxproc(),
		fade = 1+proc,
		scale = 1+(Math.abs(proc)/10);

		punchgs.TweenLite.set(revapi1077.find('.slotholder, .rs-background-video-layer'),{opacity:fade,scale:scale});		
		}
		newCall.inmodule = "parallax";
		newCall.atposition = "start";

		revapi1077.bind("revolution.slide.onloaded",function (e) {
		revapi1077.revaddcallback(newCall);
		});				
	}
}